core/src/main/scala/kafka/consumer/ConsumerFetcherManager.scala

```diff
@@ -93,8 +91,10 @@ class ConsumerFetcherManager(private val consumerIdString: String,
       }
 
       try {
-        addFetcherForPartitions(leaderForPartitionsMap.map { case (topicPartition, broker) =>
-          topicPartition -> BrokerAndInitialOffset(broker, partitionMap(topicPartition).getFetchOffset())}
+        addFetcherForPartitions(
+          leaderForPartitionsMap.map { case (topicPartition, broker) =>
+            topicPartition -> BrokerAndInitialOffset(broker, partitionMap(topicPartition).getFetchOffset())
+          }.toMap
```

Comment:
addFetcherForPartitions expects immutable.Map and leaderForPartitionsMap is mutable.Map so calling toMap will make it immutable. If I resolve TODO above, there will be no need to call toMap here.